overlays/nodejs/serverless/function.ts

```diff
@@ -111,14 +121,16 @@ export class Function extends pulumi.ComponentResource {
             entryPoint: handlerName,
             sourceArchiveBucket: this.bucket.name,
             sourceArchiveObject: this.bucketObject.name,
-            triggerHttp: true,
+            triggerHttp: !(options.triggerBucket || options.triggerTopic),
 
             availableMemoryMb: options.availableMemoryMb,
             description: options.description,
             labels: options.labels,
             project: options.project,
             region: options.region,
-            timeout: options.timeout
+            timeout: options.timeout,
+            triggerBucket: options.triggerBucket,
+            triggerTopic: options.triggerTopic
```

Comment:
In the latest TF Google provider, these properties have actually been deprecated.  We haven't pulled that in yet - but we will very soon.  Given that, I suspect we will want to avoid building on top of them here.  https://github.com/terraform-providers/terraform-provider-google/blob/master/CHANGELOG.md#1190-october-08-2018